### PR TITLE
Fixes `winnow::ascii::float` not matching `+inf` or `-inf`

### PR DESCRIPTION
**Pull request by dezyh**
Fixes `winnow::ascii::float` not matching any variants of `inf`/`infinite` prefixed by a `+`/`-` sign as mentioned in https://github.com/winnow-rs/winnow/issues/411.

### CODE REVIEW

**Comment by dezyh** [comment]
Aside, I saw this `cargo clippy` lint near my changes
```
warning: use of deprecated function `token::tag_no_case`: Replaced with `tag(ascii::Caseless(_))`
  --> src/ascii/mod.rs:18:19
   |
18 | use crate::token::tag_no_case;
   |                   ^^^^^^^^^^^
   |
   = note: `#[warn(deprecated)]` on by default
```

However, proposed solution does not work in this method so I'm not considering that lint with this PR. However, if you have any advice, let me know and I'll take a quick look.

This was the error when using `crate::token::tag` and `crate::ascii::Caseless`
```
error[E0277]: the trait bound `I: Compare<Caseless<&str>>` is not satisfied
    --> src/ascii/mod.rs:1370:9
     |
1370 |         tag(Caseless("nan")),
     |         ^^^ the trait `Compare<Caseless<&str>>` is not implemented for `I`
     |
note: required by a bound in `token::tag`
    --> src/token/mod.rs:141:17
     |
138  | pub fn tag<T, I, Error: ParserError<I>>(tag: T) -> impl Parser<I, <I as Stream>::Slice, Error>
     |        --- required by a bound in this function
...
141  |     I: Stream + Compare<T>,
     |                 ^^^^^^^^^^ required by this bound in `tag`
help: consider further restricting this bound
     |
1361 |     I: StreamIsPartial + stream::Compare<ascii::Caseless<&str>>,
     |                        ++++++++++++++++++++++++++++++++++++++++

error[E0277]: the trait bound `I: Compare<Caseless<&str>>` is not satisfied
    --> src/ascii/mod.rs:1371:35
     |
1371 |         (opt(one_of(['+', '-'])), tag(Caseless("infinity"))).recognize(),
     |                                   ^^^ the trait `Compare<Caseless<&str>>` is not implemented for `I`
     |
note: required by a bound in `token::tag`
    --> src/token/mod.rs:141:17
     |
138  | pub fn tag<T, I, Error: ParserError<I>>(tag: T) -> impl Parser<I, <I as Stream>::Slice, Error>
     |        --- required by a bound in this function
...
141  |     I: Stream + Compare<T>,
     |                 ^^^^^^^^^^ required by this bound in `tag`
help: consider further restricting this bound
     |
1361 |     I: StreamIsPartial + stream::Compare<ascii::Caseless<&str>>,
     |                        ++++++++++++++++++++++++++++++++++++++++

error[E0277]: the trait bound `I: Compare<Caseless<&str>>` is not satisfied
    --> src/ascii/mod.rs:1372:35
     |
1372 |         (opt(one_of(['+', '-'])), tag(Caseless("inf"))).recognize(),
     |                                   ^^^ the trait `Compare<Caseless<&str>>` is not implemented for `I`
     |
note: required by a bound in `token::tag`
    --> src/token/mod.rs:141:17
     |
138  | pub fn tag<T, I, Error: ParserError<I>>(tag: T) -> impl Parser<I, <I as Stream>::Slice, Error>
     |        --- required by a bound in this function
...
141  |     I: Stream + Compare<T>,
     |                 ^^^^^^^^^^ required by this bound in `tag`
help: consider further restricting this bound
     |
1361 |     I: StreamIsPartial + stream::Compare<ascii::Caseless<&str>>,
     |                        ++++++++++++++++++++++++++++++++++++++++
```

**Comment by dezyh** [comment]
In retrospect, I noticed the infinite float values are actually comparable to themselves (unlike `NaN`), so we could move all the infinity variants into the main `test_cases` which makes my refactoring changes a little redundant. 

Let me know what you think.

File: src/ascii/tests.rs [hunk 1]
```diff
@@ -575,16 +575,31 @@ mod complete {
...
-        assert!(nan.is_nan());
-        assert_eq!(i, "");
-
-        let (i, inf) = float::<_, f32, ()>.parse_peek("inf").unwrap();
-        assert!(inf.is_infinite());
-        assert_eq!(i, "");
-        let (i, inf) = float::<_, f32, ()>.parse_peek("infinity").unwrap();
-        assert!(inf.is_infinite());
-        assert_eq!(i, "");
+        let mut special_test_cases = vec!["NaN", "inf", "infinity"];
```
Comment (epage) [inline src/ascii/tests.rs]: nit: this could be a `[]` instead of `vec![]`

File: src/ascii/tests.rs [hunk 2]
```diff
@@ -575,7 +575,22 @@ mod complete {
...
+            "Infinity",
+            "-inf",
+            "-Inf",
+            "-infinity",
+            "-Infinity",
+            "+inf",
+            "+Inf",
+            "+infinity",
+            "+Infinity",
+        ];
```
Comment (epage) [inline src/ascii/tests.rs:593]: A commit should not be failing tests.  What I would expect is the commit should pass tests and the follow up commit should update the tests to show they still work.

Benefits:
- Helps test the tests, making sure you are testing the right thing
- Makes the intended behavior change clearer

File: src/ascii/mod.rs [hunk 3]
```diff
@@ -15,6 +15,7 @@ use crate::error::{ErrMode, ErrorKind, Needed};
 use crate::stream::{AsBStr, AsChar, ParseSlice, Stream, StreamIsPartial};
 use crate::stream::{Compare, CompareResult};
 use crate::token::one_of;
+use crate::token::tag_no_case;
```
Comment (epage) [inline src/ascii/mod.rs:18]: nit: this is an unrelated change, so please revert it

(notice that the function has an `allow(deprecated)` on it for `tag_no_case` because we can't resolve the deprecation until we make a breaking change)

**Comment by epage** [comment]
> In retrospect, I noticed the [infinite float values](https://github.com/winnow-rs/winnow/pull/412/files#diff-be1c8124f4111e9a2aa0e42369d9cc208120284851786b20f09e63aa47077383R581-R592) are actually comparable to themselves (_"Infinite operands of the same sign shall compare equal"_) (unlike `NaN`), so we could move all the infinity variants into the main `test_cases` which makes my refactoring changes a little redundant.
> 
> Let me know what you think.

I can go either way on this

File: src/ascii/tests.rs (hunk 1 above)
Comment (dezyh) [reply src/ascii/tests.rs]: Thanks. I saw this too but didn't want to add unnecessary changes to the PR. 
I've changed both this usage and the existing usage of `vec![]` to an array `[]`.

File: src/ascii/mod.rs (hunk 3 above)
Comment (dezyh) [reply src/ascii/mod.rs]: Apologies, I didn't realize it was the `use crate::token::tag_no_case` which was triggering the clippy deprecated lint (rather than simply calling `crate::token::tag_no_case`).

File: src/ascii/tests.rs (hunk 1 above)
Comment (epage) [reply src/ascii/tests.rs]: Ah, I had missed that `vec!` was already used
